tikv-client/src/main/java/com/pingcap/tikv/predicates/TiKVScanAnalyzer.java

```diff
@@ -498,22 +499,39 @@ public Map<Long, List<KeyRange>> buildTableScanKeyRange(
       TiTableInfo table, List<IndexRange> indexRanges, List<TiPartitionDef> prunedParts) {
     requireNonNull(table, "Table is null");
     requireNonNull(indexRanges, "indexRanges is null");
-    IntegerType handleDatatype;
-    if (table.isPkHandle()) {
-      handleDatatype = (IntegerType) table.getPKIsHandleColumn().getType();
-    } else {
-      handleDatatype = IntegerType.ROW_ID_TYPE;
-    }
+    List<Long> ids = new ArrayList<>();
     if (table.isPartitionEnabled()) {
-      List<Long> ids = new ArrayList<>();
       for (TiPartitionDef pDef : prunedParts) {
         ids.add(pDef.getId());
       }
-      return buildTableScanKeyRangeWithIds(ids, indexRanges, handleDatatype);
     } else {
-      return buildTableScanKeyRangeWithIds(
-          ImmutableList.of(table.getId()), indexRanges, handleDatatype);
+      ids.add(table.getId());
     }
+    if (table.isCommonHandle()) {
+      return buildTableCommonHandleScanKeyRange(ids, indexRanges);
+    } else {
+      IntegerType handleDatatype;
+      if (table.isPkHandle()) {
+        handleDatatype = (IntegerType) table.getPKIsHandleColumn().getType();
+      } else {
+        handleDatatype = IntegerType.ROW_ID_TYPE;
+      }
+      return buildTablePKScanKeyRangeWithIds(ids, indexRanges, handleDatatype);
+    }
+  }
+
+  private Map<Long, List<KeyRange>> buildTableCommonHandleScanKeyRange(
+      List<Long> ids, List<IndexRange> indexRanges) {
+    Map<Long, List<KeyRange>> idRanges = new HashMap<>();
+    for (long id : ids) {
+      List<KeyRange> ranges = new ArrayList<>(indexRanges.size());
+      for (IndexRange ir : indexRanges) {
+        KeyRange keyRange = new ClusterIndexScanKeyRangeBuilder(id, ir).compute();
```

Comment:
if ClusterIndexScanKeyRangeBuilder is just used in CommonHandle. alter the name like CommonHandleScanKeyRangeBuilder. if it can be used in pkisHandle, reuse it in buildTablePKScanKeyRangeWithIds

Reply:
changed